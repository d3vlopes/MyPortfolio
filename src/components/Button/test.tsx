import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Github } from '@styled-icons/boxicons-logos'

import Button from '.'

const { getByRole, getByTestId } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    renderWithTheme(
      <Button variant="primary" size="medium">
        Button
      </Button>,
    )

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#6D59A8',
      'font-size': '2.2rem',
      width: '21.5rem',
      height: '6.2rem',
    })
  })

  it('should render the variant secondary ', () => {
    renderWithTheme(<Button variant="secondary">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#83A848',
    })
  })

  it('should render the variant outline ', () => {
    renderWithTheme(<Button variant="outline">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      border: '1px solid #DEE7EA',
      'background-color': 'transparent',
    })
  })

  it('should render expand version', () => {
    renderWithTheme(<Button expand>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '31.5rem',
      height: '6.2rem',
    })
  })

  it('should render small size', () => {
    renderWithTheme(<Button size="small">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '1.8rem',
      width: '17.5rem',
      height: '5rem',
    })
  })

  it('should render large size', () => {
    renderWithTheme(<Button size="large">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '2.6rem',
      width: '28.8rem',
      height: '7.6rem',
    })
  })

  it('should render an icon version', () => {
    renderWithTheme(
      <Button icon={<Github data-testid="icon" />}>Button</Button>,
    )

    const button = getByRole('button', { name: /button/i })
    const icon = getByTestId('icon')

    expect(button).toBeInTheDocument()
    expect(icon).toBeInTheDocument()
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Button
      </Button>,
    )

    expect(screen.getByRole('link', { name: /button/i })).toHaveAttribute(
      'href',
      '/link',
    )
  })

  it('should up while hover', async () => {
    renderWithTheme(<Button hoverUp>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyleRule('transform', 'translateY(-0.5rem)', {
      modifier: ':hover',
    })
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<Button>Button</Button>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
