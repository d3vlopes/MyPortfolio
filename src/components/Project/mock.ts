import { ProjectProps } from '.'

export default [
  {
    src: '/img/projects/kurtis.png',
    name: 'Kurtis Classrom',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas felis nam nulla bibendum cras viverra fermentum vitae. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas felis nam nulla bibendum cras viverra fermentum vitae. ',
    slug: 'kurtis-classrom',
  },
  {
    src: '/img/projects/won.png',
    name: 'Won Games',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas felis nam nulla bibendum cras viverra fermentum vitae. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas felis nam nulla bibendum cras viverra fermentum vitae. ',
    slug: 'won-games',
  },
] as ProjectProps[]
