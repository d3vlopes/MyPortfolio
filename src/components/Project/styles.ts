import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import { Wrapper as HeadingStyles } from 'components/Heading/styles'

export const Wrapper = styled.div`
  display: flex;
  gap: 0 8rem;

  ${media.lessThan('large')`
    flex-direction: column;
    gap: 8rem 0;
  `}

  ${media.lessThan('large')`
    align-items: center;
  `}
`

export const ImageWrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    z-index: 1;

    div {
      box-shadow: 10px 15px 10px 2px rgba(0, 0, 0, 0.15);
    }

    img {
      width: 100%;
      border-radius: ${theme.border.radius.small};

      ${media.lessThan('medium')`
        height: 21rem;
      `}
    }

    &::before {
      content: '';
      position: absolute;
      top: -4.6rem;
      left: -4.6rem;
      max-width: 48.1rem;
      width: 100%;
      height: 24.1rem;
      background: ${theme.colors.bg.secondary.main};
      z-index: -1;

      ${media.between('medium', 'huge')`
        left: -2.6rem;
      `}

      ${media.lessThan('medium')`
        top: -20%;
        height: 20rem;
      `}
    }
  `}
`

export const ContentContainer = styled.div`
  display: flex;
  flex-direction: column;
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    line-height: 170%;
    color: ${theme.colors.complementary.main};
    margin: ${theme.spacings.xsmall} 0 ${theme.spacings.medium};
    max-width: 59.8rem;
    width: 100%;

    ${media.lessThan('medium')`
      font-size: ${theme.font.sizes.xsmall};
    `}
  `}
`

export const WrapperGrid = styled.div`
  ${({ theme }) => css`
    position: relative;
    display: flex;
    flex-direction: column;

    a {
      text-decoration: none;
      color: ${theme.colors.text.light};
      transition: color ${theme.transition.default};

      img {
        width: 100%;
        border-radius: ${theme.border.radius.medium}
          ${theme.border.radius.medium} 0 0;
      }

      &:hover {
        color: ${theme.colors.primary.main};
      }
    }

    ${HeadingStyles} {
      margin-bottom: 0;
    }
  `}
`

export const ContentGridContainer = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: ${theme.spacings.medium} ${theme.spacings.xsmall};
  `}
`

export const DescriptionGrid = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    line-height: 1.5;
    color: ${theme.colors.complementary.main};
    margin-top: ${theme.spacings.xxsmall};

    ${media.lessThan('medium')`
      display: none;
    `}
  `}
`
