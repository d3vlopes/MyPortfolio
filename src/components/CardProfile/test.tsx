import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardProfile, { CardProfileProps } from '.'

const props: CardProfileProps = {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul - RS',
  liking: 'Programação, design, música, séries e filmes',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
}

const { getByRole, getByText } = screen

describe('<CardProfile />', () => {
  it('should render CardProfile', () => {
    const { container } = renderWithTheme(<CardProfile {...props} />)

    const image = getByRole('img', { name: 'Leandro Lopes' })
    const name = getByText('Leandro Lopes')
    const age = getByText('27 anos')
    const city = getByText('Santa Cruz do Sul - RS')
    const liking = getByText('Programação, design, música, séries e filmes')
    const dream = getByText(
      'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
    )

    expect(image).toHaveAttribute('src', '/img/me.jpg')
    expect(name).toBeInTheDocument()
    expect(age).toBeInTheDocument()
    expect(city).toBeInTheDocument()
    expect(liking).toBeInTheDocument()
    expect(dream).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
