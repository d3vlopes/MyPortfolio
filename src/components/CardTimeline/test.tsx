import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardTimeline, { CardTimelineProps } from '.'

const props: CardTimelineProps = {
  title: 'CardTimeline',
  content: '<p>Loren ipsum dolor</p>',
}

const { getByRole, getByText } = screen

describe('<CardTimeline />', () => {
  it('should render CardTimeline', () => {
    const { container } = renderWithTheme(<CardTimeline {...props} />)

    const title = getByRole('heading', { name: 'CardTimeline' })
    const content = getByText('Loren ipsum dolor')
    const p = container.querySelector('p')

    expect(title).toBeInTheDocument()
    expect(content).toBeInTheDocument()
    expect(p?.tagName.toLowerCase()).toBe('p')
  })

  it('should match snapshot ', () => {
    const { container } = renderWithTheme(<CardTimeline {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
