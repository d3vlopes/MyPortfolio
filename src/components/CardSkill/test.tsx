import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Sword } from '@styled-icons/remix-fill/Sword'

import CardSkill from '.'
import mock from './mock'

const { getByTestId, getByRole, getByText } = screen

describe('<CardSkill />', () => {
  it('should render CardSkill', () => {
    renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    const icon = getByTestId('icon')
    const title = getByRole('heading', { name: /desafios/i })
    const description = getByText(
      /Estou sempre em busca de novos desafios que me tirem do meu porto seguro./i,
    )

    expect(icon).toBeInTheDocument()
    expect(title).toBeInTheDocument()
    expect(description).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
