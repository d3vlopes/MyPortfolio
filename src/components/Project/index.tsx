import Link from 'next/link'
import Image from 'next/image'

import Heading from 'components/Heading'
import Button from 'components/Button'

import * as S from './styles'

export type ProjectProps = {
  src: string
  name: string
  description: string
  slug: string
  variant?: 'list' | 'grid'
}

const Project = ({
  src,
  name,
  description,
  slug,
  variant = 'list',
}: ProjectProps) => {
  const renderList = () => {
    return (
      <S.Wrapper>
        <S.ImageWrapper data-testid="list">
          <Link href={`/project/${slug}`} passHref>
            <a>
              <Image
                src={src}
                alt={name}
                width={513}
                height={249}
                quality={100}
              />
            </a>
          </Link>
        </S.ImageWrapper>
        <S.ContentContainer>
          <Heading as="h3" size="xxxlarge">
            {name}
          </Heading>
          <S.Description>{description}</S.Description>
          <Button
            size="small"
            data-testid="button"
            as="a"
            href={`/project/${slug}`}
            hoverUp
          >
            Saber mais
          </Button>
        </S.ContentContainer>
      </S.Wrapper>
    )
  }

  const renderGrid = () => {
    return (
      <S.WrapperGrid data-testid="grid">
        <Link href={`/project/${slug}`} passHref>
          <a>
            <Image
              src={src}
              alt={name}
              width={397}
              height={292}
              quality={100}
              layout="responsive"
            />
          </a>
        </Link>
        <Link href={`/project/${slug}`} passHref>
          <a>
            <S.ContentGridContainer>
              <Heading as="h3" size="large">
                {name}
              </Heading>
              <S.DescriptionGrid>{description}</S.DescriptionGrid>
            </S.ContentGridContainer>
          </a>
        </Link>
      </S.WrapperGrid>
    )
  }

  return <>{variant === 'list' ? renderList() : renderGrid()}</>
}

export default Project
