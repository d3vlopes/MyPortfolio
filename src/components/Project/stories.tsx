import { Story, Meta } from '@storybook/react'
import Project, { ProjectProps } from '.'
import mock from './mock'

export default {
  title: 'Components/Project',
  component: Project,
  parameters: {
    backgrounds: {
      default: 'secondary-dark',
    },
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/KIQWZ47GDrrNCIxysbEqOA/My-Portfolio?node-id=769%3A1073',
    },
  },
  args: mock[1],
} as Meta<ProjectProps>

export const Default: Story<ProjectProps> = (args) => {
  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        paddingTop: '8rem',
      }}
    >
      <Project {...args} />
    </div>
  )
}
