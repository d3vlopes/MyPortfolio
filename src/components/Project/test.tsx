import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import Project, { ProjectProps } from '.'

const props: ProjectProps = {
  src: '/img/projects/kurtis.png',
  name: 'Project Name',
  description: 'Loren ipsum dolor',
  slug: 'kurtis-classrom',
}

const { getByRole, getByText, getByTestId } = screen

describe('<Project />', () => {
  it('should render Project', () => {
    renderWithTheme(<Project {...props} />)

    const image = getByRole('img', {
      name: /project name/i,
    })
    const name = getByRole('heading', { name: /project name/i })
    const description = getByText('Loren ipsum dolor')
    const button = getByTestId('button')

    expect(image).toHaveAttribute('src', '/img/projects/kurtis.png')
    expect(name).toBeInTheDocument()
    expect(description).toBeInTheDocument()
    expect(button).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<Project {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
