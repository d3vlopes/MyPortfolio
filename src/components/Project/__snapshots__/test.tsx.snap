// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project /> should match snapshot 1`] = `
.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: Gilroy-Bold;
  max-width: 100%;
  font-size: 3.2rem;
  line-height: calc(3.2rem + 1rem);
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  -webkit-transition: background 0.1s ease-in-out,-webkit-transform 0.3s ease-in-out;
  -webkit-transition: background 0.1s ease-in-out,transform 0.3s ease-in-out;
  transition: background 0.1s ease-in-out,transform 0.3s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border-radius: 0.6rem;
  background-color: #6D59A8;
  font-size: 1.8rem;
  width: 17.5rem;
  height: 5rem;
}

.c6:focus {
  outline: 1px dashed #DEE7EA;
}

.c6:hover {
  -webkit-transform: translateY(-0.5rem);
  -ms-transform: translateY(-0.5rem);
  transform: translateY(-0.5rem);
}

.c6:hover svg {
  color: #6D59A8;
}

.c6 span {
  font-family: Rubik,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 500;
  line-height: 100%;
  color: #FFFFFF;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c6:hover {
  background-color: #543F8C;
}

.c6 svg {
  width: 2.2rem;
  height: 2.2rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0 8rem;
}

.c1 {
  position: relative;
  z-index: 1;
}

.c1::before {
  content: '';
  position: absolute;
  top: -4.6rem;
  left: -4.6rem;
  max-width: 48.1rem;
  width: 100%;
  height: 24.1rem;
  background: #3B3A51;
  z-index: -1;
}

.c2 {
  width: 100%;
  max-width: 51.3rem;
  box-shadow: 10px 15px 10px 2px rgba(0,0,0,0.15);
  border-radius: 0.4rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  font-size: 1.8rem;
  line-height: 170%;
  color: #DEE7EA;
  margin: 1.6rem 0 3.2rem;
  max-width: 59.8rem;
  width: 100%;
}

@media (max-width:1170px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 8rem 0;
  }
}

@media (max-width:1170px) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:768px) and (max-width:1440px) {
  .c1::before {
    left: -2.6rem;
  }
}

@media (max-width:768px) {
  .c1::before {
    top: -20%;
    height: 20rem;
  }
}

@media (max-width:768px) {
  .c2 {
    height: 21rem;
  }
}

@media (max-width:768px) {
  .c5 {
    font-size: 1.6rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    data-testid="list"
  >
    <a
      href="/project/kurtis-classrom"
    >
      <img
        alt="Project Name"
        class="c2"
        src="/img/projects/kurtis.png"
      />
    </a>
  </div>
  <div
    class="c3"
  >
    <h3
      class="c4"
    >
      Project Name
    </h3>
    <p
      class="c5"
    >
      Loren ipsum dolor
    </p>
    <a
      class="c6"
      data-testid="button"
      href="/project/kurtis-classrom"
    >
      <span>
        Saber mais
      </span>
    </a>
  </div>
</div>
`
